lib/tools/apk-utils.js

```diff
@@ -472,16 +472,21 @@ apkUtilsMethods.install = async function install (appPath, options = {}) {
     timeoutCapName: 'androidInstallTimeout',
   }, options);
 
-  const cachedRemotePath = await this.cacheApk(appPath, {
-    timeout: options.timeout,
-  });
   const installArgs = buildInstallArgs(await this.getApiLevel(), options);
-  try {
-    const started = process.hrtime();
-    const output = await this.shell(['pm', 'install', ...installArgs, cachedRemotePath], {
+  const installOpts = {
+    timeout: options.timeout,
+    timeoutCapName: options.timeoutCapName,
+  };
+  let installMethod = async () => await this.adbExec(['install', ...installArgs, appPath], installOpts);
+  if (this.remoteAppsCacheLimit > 0) {
```

Comment:
Maybe just add a comment here to clarify that `remoteAppCacheLimit <= 0` means to not cache apk on devices.

Reply:
I try to write code the way it does not need explanation comments, but if you think it is still necessary...

Reply:
I'm not a stickler for having comments _everywhere_ but in this case, I just want an explanation of why the caching behaviour is different for `this.remoteAppsCacheLimit > 0` vs. `this.remoteAppsCacheLimit <= 0`.